docs/DeveloperGuide.md

```diff
@@ -441,17 +509,68 @@ Currently, names in Foster Family must be alphanumeric. However, this excludes c
 characters such as `/`. For example, we currently do not allow  `s/o` in a person's name as the `/` is used as a command
 delimiter. Hence, one possible improvement is to enforce that the name inputted by the user must be enclosed in quotation marks for parsing, and to allow symbols such as `/`, `'`, `-` etc. using regex. Additionally, we will disallow the use of numeric values in names, to prevent the case where a number is inputted as a name.
 
-e.g. `name/"Henry Tan"` and `name/"Nagaratnam s/o Suppiah"` are now valid name parameters.
+e.g. `n/"Henry Tan"` and `n/"Nagaratnam s/o Suppiah"` are now valid name parameters.
 
 
 ### Phone Number Input
 
+Currently, phone numbers in Foster Family accept more than 8 digits as a valid input. This might cause invalid phone numbers to 
+be recorded without error messages to warn the user of such mistakes, especially in the situation where local phone numbers are used.
+Hence, one possible improvement is to enforce that the phone number inputted by the user must be restricted to a maximum of 15 
+digits (according to the international phone numbering plan).
+
+e.g. `p/90876534567890234567` is now an invalid phone number parameter.
+
 ### Handle Invalid Fosterer - Corrupt Data File
 
+Currently, the Foster Family data is saved automatically as a JSON file, and in the case where the data file is updated directly
+and made invalid, Foster Family would either discard all data and start with an empty data file at the next run or the app would 
+not start at all. Even though it is advised in the user guide that user should not make changes to the data file directly, one 
+possible improvement to be made to prevent such incidents would be to either use a database with username and password authentication 
+or implement encryption.
```

Comment:
I think this may not solve the issue of them editing the file and getting an invalid fosterer! i think what we can suggest is to parse the data file(?) and if got any data inconsistencies / formatting error, we will alert the user the user which fosterer/ which line the error is in, and ask them to rectify the error before retrying. so this way, the data file will not get wiped. (idk HAHAH)

Reply:
eh idk how we can do this HAHA

Reply:
added a potential solution